### PR TITLE
[core] WIP: Extending ACK with RCV drop total

### PR DESCRIPTION
**Pull request by maxsharabayko**
**Problem:** Sender does not know if and how many packets are being dropped by the receiver. SRT receiver, dropping a packet, just sends a regular ACK control packet, as if it had received them.

**Proposed Solution**: Extend ACK control packet to contain information about the number of packets dropped by the receiver: "Total Dropped Packets Count" field.

```
    0                   1                   2                   3
    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
   +-+-+-+-+-+-+-+-+-+-+-+-+- SRT Header +-+-+-+-+-+-+-+-+-+-+-+-+-+
0  |1|        Control Type         |           Reserved            |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
1  |                    Acknowledgement Number                     |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
2  |                           Timestamp                           |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
3  |                     Destination Socket ID                     |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+- CIF -+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
4  |            Last Acknowledged Packet Sequence Number           |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
5  |                              RTT                              |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
6  |                          RTT Variance                         |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
7  |                     Available Buffer Size                     |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
8  |                     Packets Receiving Rate                    |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
9  |                     Estimated Link Capacity                   |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
10 |                         Receiving Rate                        |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
11 |                   Total Dropped Packets Count                 | <-- PROPOSED FIELD!!!
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```

**Total Dropped Packets Count** - the total number of packets dropped by the receiver
from the start of receiving. Only sent if differs from the previous value,
sent in previous ACK packets, that were acknowledged by the sender (ACKACK).
In other words, included in ACK packet until ACKACK is received acknowledging that the sender has received this number of dropped packets.

Older SRT versions will ignore this field (maximum size is not restricted). Also receiver knows sender’s SRT version, and can only send an extended ACK packet for newer versions.
ACKACK control packet: no changes are required. The structure remains the same.

### TODO

- [ ] Add API statistics (note that extending statistics API breaks ABI compatibility between versions).
- [ ] Do not include the **Total Dropped Packets Count** field if the value hasn't changed and the previous value was already received by sender (receiver got ACACK for ACK with that value).
- [ ] Handle possible `uint32_t` overflow on the sender side.

### CODE REVIEW

**Comment by ethouris** [comment]
Why can't this be simply the number of dropped packets among those being ACK-ed with this control packet?

**Comment by maxsharabayko** [comment]
> Why can't this be simply the number of dropped packets among those being ACK-ed with this control packet?

To handle the cases when ACK packet is lost.
Sending the total number allows recovering the actual loss value of the receiver with the first arrived ACK.

**Comment by mbakholdina** [comment]
@maxsharabayko Please take into consideration Issue #1001 when proceeding with the current PR.
